packages/nimble-components/src/dialog-base/index.ts

```diff
@@ -0,0 +1,108 @@
+import { attr } from '@microsoft/fast-element';
+import { FoundationElement } from '@microsoft/fast-foundation';
+import { UserDismissed } from '../patterns/dialog/types';
+
+/**
+ * This is a workaround for an incomplete definition of the native dialog element. Remove when using Typescript >=4.8.3.
+ * https://github.com/microsoft/TypeScript/issues/48267
+ * @internal
+ */
+export interface ExtendedDialog extends HTMLDialogElement {
+    showModal(): void;
+    close(): void;
+}
+
+/**
+ * A base class for components based on a native dialog.
+ */
+// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+export abstract class DialogBase<CloseReason = void> extends FoundationElement {
```

Comment:
I'd probably frame this as "Modal" of which the Dialog and Drawer are implementations of. But like the other comment says haven't fully thought about what the specific responsibilities of the Modal base class should be